product_end_of_life/views/purchase_view.xml

```diff
@@ -5,9 +5,12 @@
         <field name="model">purchase.order</field>
         <field name="inherit_id" ref="purchase.purchase_order_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//sheet//notebook//page//field//tree//field[@name='product_id']"
+            <xpath expr="//sheet//notebook//page//field[@name='order_line']//tree//field[@name='product_id']"
```

Comment:
Please change from `<xpath expr="//sheet//notebook//page//field[@name='order_line']//tree//field[@name='product_id']" ` to `<xpath expr="//field[@name='order_line']//field[@name='product_id']"` as @simahawk  mentioned. ;-)